packages/server/lambdas/alfresco/list-files.ts

```diff
@@ -62,20 +47,45 @@ const searchByTermWithParent = async (
     });
     const alfrescoSearchAPIUrl = `${getAlfrescoUrlBase()}/search/versions/1/search`;
     const options = await getAlfrescoOptions(uid, { 'Content-Type': 'application/json;charset=UTF-8' });
-
     const response = await axios.post(`${alfrescoSearchAPIUrl}`, bodyRequest, options);
     return response.data;
   } catch (err) {
     throw err;
   }
 };
 
+const getFolder = async (uid: string, nodeId: string) => {
+  try {
+    const alfrescoCoreAPIUrl = `${getAlfrescoUrlBase()}/alfresco/versions/1`;
+    const url = `${alfrescoCoreAPIUrl}/nodes/${nodeId}?where=(isFolder=true)&include=path`;
+    const options = await getAlfrescoOptions(uid, { 'Content-Type': 'application/json;charset=UTF-8' });
+    const response = await axios.get(url, options);
+    return response.data;
+  } catch (error: any) {
+    if (error instanceof AxiosError) {
+      // In case nodeId doesn't exist, Alfresco throws 404
+      if (error.response?.status === 404) {
+        return null;
+      }
+    }
+    throw error;
+  }
+};
+
+const isFolderInCategory = async (folderPath: string, category: string) => {
+  // Split the path into its components
+  const pathComponents = folderPath.split('/');
+
+  // Check if the parent folder name is among the path components
+  return pathComponents.includes(category);
```

Comment:
Can this be specified to a certain index? Because otherwise, if we later on add some hidden archive folder, which could have something like `/archive/linjakaaviot/subfolder`, then this check would return true if you're looking for the subfolder and claim that it's in the "normal" linjakaaviot. Bit of a far fetch, as the user would need to know the id of the subfolder, but it's a possibility.